ext/oci8/oci8_interface.c

```diff
@@ -44,6 +44,29 @@
 #define OCI_STMT_CALL 10
 #endif
 
+/* {{{ proto bool oci_register_taf_callback( resource connection [, string callback] )
+   Register a callback function for Oracle Transparent Application Failover (TAF) */
+PHP_FUNCTION(oci_register_taf_callback)
+{
+	zval *z_connection;
+	php_oci_connection *connection;
+	char *callback = NULL;
+	int callback_len = 0;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r|s!", &z_connection, &callback, &callback_len) == FAILURE) {
```

Comment:
Right, I just didn't know how to properly store the `zend_fcall_info` for later. Also a callback rarely occurs so it probably would be better to store the `zend_fcall_info` without the cache.

I'm fairly new to PHP extension writing and still don't fully understand the `zval` reference counter; as the `zend_fcall_info` contains multiple `zval`, would I have to increase all of them? And do I actually have to increase the counter after receiving the `zend_fcall_info`? It's hard to find answers to such questions - or maybe I'm just looking at the wrong locations ^^